src/time-range-picker/time-range-picker.tsx

```diff
@@ -0,0 +1,220 @@
+    placeholder,
+    format = "24hr",
+    readOnly,
+    onChange,
+    onBlur,
+    onSelectionCancel,
+    ...otherProps
+}: TimeRangePickerProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [showStartTimeSelector, setShowStartTimeSelector] =
+        useState<boolean>(false);
+    const [showEndTimeSelector, setShowEndTimeSelector] =
+        useState<boolean>(false);
+    const [startTimeVal, setStartTimeVal] = useState("");
+    const [endTimeVal, setEndTimeVal] = useState("");
+
+    const nodeRef = useRef<HTMLDivElement>();
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        if (value) {
+            setStartTimeVal(value.startTime);
+            setEndTimeVal(value.endTime);
+        }
+    }, []);
+    useEffect(() => {
+        document.addEventListener("mousedown", handleMouseDownEvent);
+        document.addEventListener("keyup", handleKeyUpEvent);
+
+        return () => {
+            document.removeEventListener("mousedown", handleMouseDownEvent);
+            document.removeEventListener("keyup", handleKeyUpEvent);
+        };
+    }, [showEndTimeSelector]);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const handleStartTimeFocus = useCallback(() => {
+        if (!disabled && !readOnly && !showStartTimeSelector) {
+            setShowEndTimeSelector(false);
+            setShowStartTimeSelector(true);
+        }
+    }, [showStartTimeSelector]);
+
+    const handleEndTimeFocus = useCallback(() => {
+        if (!disabled && !readOnly && !showEndTimeSelector) {
+            setShowStartTimeSelector(false);
+            setShowEndTimeSelector(true);
+        }
+    }, [showEndTimeSelector]);
+
+    const handleMouseDownEvent = (event: MouseEvent) => {
+        if (!disabled) {
+            runOutsideFocusHandler(event);
+        }
+    };
+
+    const handleKeyUpEvent = (event: KeyboardEvent) => {
+        switch (event.code) {
+            case "Tab":
+                runOutsideFocusHandler(event);
+                break;
+            default:
+                break;
+        }
+    };
+
+    const handleSelectionDropdownCancel = () => {
+        setShowEndTimeSelector(false);
+        setShowStartTimeSelector(false);
+        onSelectionCancel && onSelectionCancel();
+    };
+
+    const handleStartTime = (value: string) => {
+        setShowStartTimeSelector(false);
+        setShowEndTimeSelector(true);
+        setStartTimeVal(value);
+
+        const timeValue: TimeRangePickerProps = {
+            startTime: startTimeVal,
```

Comment:
the `startTimeVal` variable has not been updated in this callback yet. same issue for the end time callback
```suggestion
            startTime: value,
```